File: PYTHON/TP3/test/test_etudiant.py
from src.etudiant import Etudiant
from src.cours import Cours

def test_creation_etudiant():
    maths = Cours("Mathématiques", "M.Dupond")
    français = Cours("Français", "M.Leroy")
    cours = [maths, français]
    etudiant = Etudiant("Bob", 22, "E12345", 15.5, cours)
    assert etudiant.get_nom() == "Bob"
    assert etudiant.get_age() == 22
    assert etudiant.numero_etudiant == "E12345"
    assert etudiant.get_moyenne() == 15.5
    assert etudiant.cours == cours

def test_afficher_details_etudiant():
    e = Etudiant("Bob", 22, "E123", 15.5, [])
    result = e.afficher_details()
    assert "Bob" in result
    assert "15.5" in result

File: TP1/JAVA/TP3/test/TestEtudiant.java
package test;
import src.Etudiant;
import src.Cours;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import java.util.List;

public class TestEtudiant {

    @Test
    void test_creation_etudiant() {

        Cours maths = new Cours("Mathématiques", "M.Dupond");
        Cours francais = new Cours("Français", "M.Leroy");

        List<Cours> cours = List.of(maths, francais);

        Etudiant e = new Etudiant("Bob", 22, "E12345", 15.5, cours);

        assertEquals("Bob", e.getNom());
        assertEquals(22, e.getAge());
        assertEquals("E12345", e.getNumeroEtudiant());
        assertEquals(15.5, e.getMoyenne());
        assertEquals(cours, e.getCours());
    }

    @Test
    void test_afficher_details_etudiant() {
        Etudiant e = new Etudiant("Bob", 22, "E123", 15.5, List.of());
        e.afficherDetails(); 
    }
}